src/tokenizer.ts

```diff
@@ -432,26 +441,48 @@ export class Tokenizer {
                 break;
             // String
             case '"':
-                while (this.peek() != '"' && this.peek() != '\n' && !this.isAtEnd()) {
-                    this.advance();
-                }
-                if (this.peek() === '\n' || this.isAtEnd()) {
-                    throw new TokenizerErrors.UnterminatedStringError(this.line, this.col, this.source, this.start, this.current);
-                }
-                // Consume closing "
-                this.advance();
-                this.addStringToken(TokenType.STRING);
-                break;
-            case '\'':
-                while (this.peek() != '\'' && this.peek() != '\n' && !this.isAtEnd()) {
+            case "'":
+                let quote = c;
+                if (this.peek() == quote) { // handle multi-line string
+                    this.advance(); // second quote found and consumed
+                    if (this.peek() != quote) { // empty string ""
+                        this.addStringToken(TokenType.STRING);
+                        break;
+                    } else {
+                        this.advance(); // third quote consumed
+                    }
+                    while (this.peek() != quote && !this.isAtEnd()) {
+                        this.advance(); // advance until ending quote found
+                    }
+                    if (this.isAtEnd()) {
+                        throw new TokenizerErrors.UnterminatedStringError(this.line,
+                            this.col, this.source, this.start, this.current);
+                    }
+                    this.advance(); // consume first ending quote
+                    if (this.peek() == quote) {
+                        this.advance(); // consume second ending quote
+                    } else {
+                        throw new TokenizerErrors.UnterminatedStringError(this.line,
+                            this.col, this.source, this.start, this.current);
+                    }
+                    if (this.peek() == quote) {
+                        this.advance(); // consume third ending quote
+                        this.addMultiLineStringToken(TokenType.STRING);
+                    } else {
+                        throw new TokenizerErrors.UnterminatedStringError(this.line,
+                            this.col, this.source, this.start, this.current);
+                    }
+                } else { // other case, single-line string
```

Comment:
You can remove the else here by adding a break in the previous line to end the if. That removes one layer of nesting.